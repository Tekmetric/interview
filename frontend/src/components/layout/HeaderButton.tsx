import React from 'react';

interface HeaderButtonProps {
  onClick: () => void;
  children: React.ReactNode;
}

export const HeaderButton = ({ onClick, children }: HeaderButtonProps) => {
  return (
    <button
      className="block w-full text-left px-4 py-2 text-sm text-gray-700 hover:bg-gray-100"
      onClick={onClick}
    >
      {children}
    </button>
  );
};
