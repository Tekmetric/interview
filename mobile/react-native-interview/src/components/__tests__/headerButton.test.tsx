import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { Text } from 'react-native';
import HeaderButton from '../buttons/headerButton';

describe('HeaderButton', () => {
  it(`renders correctly`, () => {
    const tree = render(
      <HeaderButton onPress={() => console.log('test')}>
        <Text>snapshot</Text>
      </HeaderButton>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('calls onPress when pressed', () => {
    const onPressMock = jest.fn();
    const { getByTestId } = render(
      <HeaderButton onPress={onPressMock}>
        <Text>press me</Text>
      </HeaderButton>,
    );

    fireEvent.press(getByTestId('header-button'));
    expect(onPressMock).toHaveBeenCalled();
  });
});
