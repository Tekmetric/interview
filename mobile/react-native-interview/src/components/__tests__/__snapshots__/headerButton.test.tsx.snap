// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderButton renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "marginHorizontal": 16,
      },
      undefined,
    ]
  }
  testID="header-button"
>
  <Text>
    snapshot
  </Text>
</View>
`;
